packages/container/lib/container.js

```diff
@@ -191,15 +280,47 @@ function lookup(container, fullName, options = {}) {
     return container.cache[fullName];
   }
 
-  let value = instantiate(container, fullName);
+  return instantiateFactory(container, fullName, options);
```

Comment:
So this means`container.lookup` is now based on the the non-double-extended factory? This seems like the behavior we didn't intend to change until `_lookupFactory` is removed.